clang-tools-extra/test/clang-tidy/checkers/readability/redundant-control-flow.cpp

```diff
@@ -222,3 +222,33 @@ void call_templates() {
   template_loop(10L);
   template_loop(10U);
 }
+
+void dont_delete_lines_before_return_statement() {
+	do {} while (0);
+#ifdef FOO
+#endif
+  return;
+}
+// CHECK-MESSAGES: :[[@LINE-2]]:3: warning: redundant return statement
+// CHECK-FIXES:      void dont_delete_lines_before_return_statement() {
+// CHECK-FIXES-NEXT: 	 do {} while (0);
+// CHECK-FIXES-NEXT: #ifdef FOO
+// CHECK-FIXES-NEXT: #endif
+// CHECK-FIXES-NEXT: }
+
+void dont_delete_lines_before_continue_statement() {
+  for (;;) {
+	  do {} while (0);
+#ifdef BAR
+#endif
+    continue;
```

Comment:
Will be good idea to add tests where semicolon is separated with spaces and located on other line.